addons/payment_stripe/models/payment_acquirer.py

```diff
@@ -73,8 +154,163 @@ def _stripe_make_request(self, endpoint, payload=None, method='POST', offline=Fa
+            'company[address][postal_code]': self.company_id.zip or '',
+            'company[address][state]': self.company_id.state_id.name or '',
+            'company[name]': self.company_id.name,
+            'individual[address][city]': self.company_id.city or '',
+            'individual[address][country]': self.company_id.country_id.code or '',
+            'individual[address][line1]': self.company_id.street or '',
+            'individual[address][line2]': self.company_id.street2 or '',
+            'individual[address][postal_code]': self.company_id.zip or '',
+            'individual[address][state]': self.company_id.state_id.name or '',
+            'individual[email]': self.company_id.email or '',
+            'business_profile[name]': self.company_id.name,
+        }
+
+    def _stripe_account_link_url(self, connected_account_id, menu_id):
+        """ Create an account link and return its URL.
+
+        An account link url is the beginning URL of Stripe Onboarding.
+        This URL is only valid once, and can only be used once.
+
+        :param str connected_account_id: The id of the connected account
+        :param int menu_id: The id of the menu to return to after onboarding.
+        :return: The account link URL
+        :rtype: str
+        """
+        return_url = urls.url_join(
+            self.company_id.get_base_url(),
+            StripeController._onboarding_return_url + f'{self.id}/{menu_id}'
+        )
+        refresh_url = urls.url_join(
+            self.company_id.get_base_url(),
+            StripeController._onboarding_refresh_url + f'{self.id}/{connected_account_id}/{menu_id}'
+        )
+        account_link = self._stripe_make_proxy_request('account_links', payload={
+            'account': connected_account_id,
+            'return_url': return_url,
+            'refresh_url': refresh_url,
+            'type': 'account_onboarding',
+        })
+        return account_link['url']
+
+    def _stripe_make_proxy_request(self, endpoint, payload=None, version=1):
+        """ Make a request to the Stripe proxy at the specified endpoint.
+
+        :param str endpoint: The proxy endpoint to be reached by the request
+        :param dict payload: The payload of the request
+        :param int version: The proxy version used
+        :return The JSON-formatted content of the response
+        :rtype: dict
+        :raise: ValidationError if an HTTP error occurs
+        """
+        proxy_payload = {
+            'jsonrpc': '2.0',
+            'id': uuid.uuid4().hex,
+            'method': 'call',
+            'params': {'payload': payload},
+        }
+        proxy_url = self.env['ir.config_parameter'].sudo().get_param(
+            'payment_stripe.fallback_proxy_url', FALLBACK_PROXY_URL
+        )
+        url = urls.url_join(proxy_url, f'{version}/{endpoint}')
+        try:
+            response = requests.post(url=url, json=proxy_payload, timeout=60)
```

Comment:
can requests really take up to 60 seconds ?

Reply:
The proxy implements a method `_stripe_make_request` similar to the one we can find in this file and has the same timeout (60). So it handles the case where Stripe is slow and takes up to 60 seconds.

We may change it, but I guess we should do it also in both implementation of `_stripe_make_request` (proxy and here in stripe/models/payment_acquirer.py)

Reply:
I don't really see an issue with a 60 seconds timeout. The request goes through the proxy, then Stripe, then the proxy again, then the db. If a user waits a full minute before onboarding their Stripe account it's not a big deal. However, if we lower the timeout to 10 seconds and someday IAP decides to be slow, everyone's onboarding is ruined.

Reply:
The issue with too long requests is that it blocks a worker on the server (we have only like 10-20 on a server). If the server never replies, during a minute, the visitors have to share one worker. Imagine multiple people make payment at the same time (e.g. odoo band tickets :guitar: ), it could be an issue. The question is "if the server does not answer after X seconds, what are the chances it will answer after Y ?")